lib/cli/eventually-list-version.js

```diff
@@ -22,9 +24,39 @@ module.exports = async () => {
     if (!cliParams.has('-v')) return false;
   }
 
+  const localServerlessPath = await resolveLocalServerlessPath();
+
+  if (localServerlessPath) {
+    // If the version is already local, do not try to fallback for version resolution to avoid falling into the loop
```

Comment:
Let's also add a comment:

```
// TODO: Remove local version fallback with next major (when its moved to the top of the process)
```

(When releasing a new major I'd search for all TODO comments in codebase and ensure we've addressed them)

I'd add also this comment to error handler